operators/wireguard/internal/controllers/cluster-connection/controller.go

```diff
@@ -141,6 +147,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return step.ReconcilerResponse()
 	}
 
+	if step := r.reconCoredns(req); !step.ShouldProceed() {
+		return step.ReconcilerResponse()
+	}
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_refinement):** Consider encapsulating rollout logic into a method.

The logic to determine if a rollout is needed and to perform the rollout is significant. Encapsulating this into a separate method could improve readability and reusability.

```suggestion
func (r *ClusterConnectionReconciler) needsRollout(ctx context.Context, obj *v1.ClusterConnection, secBytes []byte) (bool, error) {
	s, err := r.Api.Get(ctx, r.Client, fn.NN(ResourceNamespace, fmt.Sprintf("%s-gateway-configs", obj.Name)), &corev1.Secret{})
	if err != nil {
		return false, err
	}
	return strings.TrimSpace(string(secBytes)) != strings.TrimSpace(string(s.Data["server-config"])), nil
}
```